docs/examples/linux_kernel.md

```diff
@@ -0,0 +1,99 @@
+# Comparing Linux kernels
+
+The main focus of DiffKemp is on Red Hat Enterprise Linux (RHEL), whose kernel
+includes a list of functions (kABI symbols) whose semantics should remain the
+same across the lifetime of a single major release. In this example, we will
+show how to compare two versions of the kernel. If you want to try this on your
+own, be aware that it can take a few hours to get the results.
+
+## 1. Downloading and preparing the kernels
+
+Because getting and comparing the kernels requires additional dependencies
+we will use [Nix development environment](../development.md#nix-as-development-environment).
+After entering the development shell using `nix develop` command, we need to
+build DiffKemp. Then we can use [`rhel-kernel-get`](https://github.com/viktormalik/rhel-kernel-get)
+script for downloading and preparing the kernels. We will compare RHEL/CentOS
+8.2 (`4.18.0-193.el8`) and 8.3 (`4.18.0-240.el8`) versions, which we can get by:
+
+```sh
+mkdir -p kernel
+rhel-kernel-get --kabi 4.18.0-193.el8 -o kernel
+rhel-kernel-get --kabi 4.18.0-240.el8 -o kernel
+```
+
+The `--kabi` option specifies that we want to download the list of functions
+whose semantics should remain the same, and the `-o kernel` specifies the folder
+to which we want to download the kernels. After the kernels are downloaded and
+prepared, we can create *snapshots* from them and compare them.
+
+## 2. Creating snapshots of the kernels
+
+For creating snapshots of the kernels we won't be using `diffkemp build`
+but `diffkemp build-kernel` command which is specifically made for the kernel:
+
+```sh
+mkdir -p snapshots
+diffkemp build-kernel kernel/linux-4.18.0-193.el8/ snapshots/linux-4.18.0-193.el8/ kernel/linux-4.18.0-193.el8/kabi_whitelist_x86_64
+diffkemp build-kernel kernel/linux-4.18.0-240.el8/ snapshots/linux-4.18.0-240.el8/ kernel/linux-4.18.0-240.el8/kabi_whitelist_x86_64
+```
+
+The `kernel/linux-4.18.0-XXX.el8/` directories are directories with
+the downloaded and prepared 8.2 and 8.3 kernels,
+the `snapshots/linux-4.18.0-XXX.el8/` are directories where the created
+snapshots will be saved and `kerne/linux-4.18.0-XXX.el8/kabi_whitelist_x86_64`
+files are files that contain the list of kABI symbols.
```

Comment:
I think that kABI should be upper-case since it's an abbreviation.

Reply:
Sure, I changed it. Originally I wrote it like that because I saw it written like that [here](https://access.redhat.com/solutions/444773).